### PR TITLE
memkind_destroy_kind fix

### PR DESCRIPTION
**Pull request by michalbiesek**
update partition_map in case of destroy kind
Ref: #57
Besides changes described in issue - cleaning memkind_registry_g.partition_map, there are two additonal changes:
-in memkind_arena_create_map after calling arenas.create - jemalloc could return arena_index in ascending or descending order - the change here is need to detect both cases  (test_TC_MEMKIND_PmemDestroyKindArenaZero cover this case )
-munmap must be called in pmem_extent_dalloc to unmap created arenas and in pmem_extent_destroy to unmap memory maped by memkind_pmem_mmap 

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/64)

### CODE REVIEW

**Comment by michalbiesek** [comment]
___
*[src/memkind.c, line 436 at r1](https://reviewable.io/reviews/memkind/memkind/64#-LIkBQhE0cugYaB4WYGd:-LIkjKN52anviQmeqRL5:b-a4td1k) ([raw file](https://github.com/memkind/memkind/blob/eacb2f68954739a58264ff0a19da90312bc5df3d/src/memkind.c#L436)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

maybe 2 conditions like memkind_destroy_kind above, unnecessary space after parenthesis
</blockquote></details>

I change logic a little bit

**Comment by michalbiesek** [comment]
___
*[src/memkind.c, line 438 at r1](https://reviewable.io/reviews/memkind/memkind/64#-LIkVgpSDyKs2Y_NChu3:-LIkjUdV5OGqZRXgqtJZ:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/eacb2f68954739a58264ff0a19da90312bc5df3d/src/memkind.c#L438)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

*exists
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[src/memkind.c, line 442 at r1](https://reviewable.io/reviews/memkind/memkind/64#-LIjtI4h6BlQREY4-uAu:-LIkjXtKD3lj4UCHfi7E:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/eacb2f68954739a58264ff0a19da90312bc5df3d/src/memkind.c#L442)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

} else {
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[src/memkind.c, line 459 at r1](https://reviewable.io/reviews/memkind/memkind/64#-LIkZkqZCgfv7L6M3XH7:-LIkjYpcBuxUimkpcD3r:b8xtxd6) ([raw file](https://github.com/memkind/memkind/blob/eacb2f68954739a58264ff0a19da90312bc5df3d/src/memkind.c#L459)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

I'm not sure is a good practice to put code as an error message
</blockquote></details>

Done - I remove it.

**Comment by michalbiesek** [comment]
___
*[src/memkind.c, line 741 at r1](https://reviewable.io/reviews/memkind/memkind/64#-LIk_TEeENEK7un-a8z9:-LIkjcY7CBvIXrCNAGT2:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/eacb2f68954739a58264ff0a19da90312bc5df3d/src/memkind.c#L741)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

} else {
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[src/memkind.c, line 315 at r2](https://reviewable.io/reviews/memkind/memkind/64#-LIktn8h8w1FW_xOx9qh:-LIocRXl18e5SmOjsWBJ:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/0ee65863530b58c219495a9f319592c87987f602/src/memkind.c#L315)):*
<details><summary><i>Previously, kasiawasiuta wrote…</i></summary><blockquote>

Consider splitting in two lines
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[test/memkind_pmem_tests.cpp, line 172 at r2](https://reviewable.io/reviews/memkind/memkind/64#-LIoK5veAyVb0utkCHFG:-LIocV7jBM-vhNkNBm8-:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/0ee65863530b58c219495a9f319592c87987f602/test/memkind_pmem_tests.cpp#L172)):*
<details><summary><i>Previously, kasiawasiuta wrote…</i></summary><blockquote>

Pls add whitespace before "="
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[test/memkind_pmem_tests.cpp, line 180 at r2](https://reviewable.io/reviews/memkind/memkind/64#-LIoK1gZ3j7MTVm653-n:-LIocWyl6skUgcgNnISf:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/0ee65863530b58c219495a9f319592c87987f602/test/memkind_pmem_tests.cpp#L180)):*
<details><summary><i>Previously, kasiawasiuta wrote…</i></summary><blockquote>

++i
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[test/memkind_pmem_tests.cpp, line 197 at r2](https://reviewable.io/reviews/memkind/memkind/64#-LIoKHgo3lxsl-nbr_Ja:-LIocYb8DQZjmoYjwgrv:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/0ee65863530b58c219495a9f319592c87987f602/test/memkind_pmem_tests.cpp#L197)):*
<details><summary><i>Previously, kasiawasiuta wrote…</i></summary><blockquote>

Add whitespace
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[src/memkind.c, line 315 at r2](https://reviewable.io/reviews/memkind/memkind/64#-LIktn8h8w1FW_xOx9qh:-LL9BnE82Ua6MYAiNhjZ:bb6un1j) ([raw file](https://github.com/memkind/memkind/blob/0ee65863530b58c219495a9f319592c87987f602/src/memkind.c#L315)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

If you divided lines you can remove unnecessary brackets. table[i] is not an expression.
Such if statement: 
if(a == 1 && 
   b == 0)  
is clear enough
</blockquote></details>

Done

**Comment by michalbiesek** [comment]
___
*[src/memkind_arena.c, line 476 at r3](https://reviewable.io/reviews/memkind/memkind/64#-LL8vPKN9nCp_KVww-DC:-LL9BrMbAK7H_r3VJO3d:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/25955d2fbc511982125af6d0c2a6cb95669371f2/src/memkind_arena.c#L476)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Unnecessary line spacing
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[test/memkind_pmem_tests.cpp, line 172 at r2](https://reviewable.io/reviews/memkind/memkind/64#-LIoK5veAyVb0utkCHFG:-LL9BxHF6qlRrLbWbnpn:bb6un1j) ([raw file](https://github.com/memkind/memkind/blob/0ee65863530b58c219495a9f319592c87987f602/test/memkind_pmem_tests.cpp#L172)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

And remove one space around "*"
</blockquote></details>

Done

**Comment by michalbiesek** [comment]
___
*[test/memkind_pmem_tests.cpp, line 266 at r4](https://reviewable.io/reviews/memkind/memkind/64#-LLTCwWb6Jx0UN3AbmcM:-LLTPS0ADgvoAxue6A8z:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/074d77c9104c79e116ed3d144f22d3028a8cfc50/test/memkind_pmem_tests.cpp#L266)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Detail: space before closing bracket in whole file
</blockquote></details>

Done.

File: src/memkind_arena.c [hunk 1]
```diff
@@ -468,6 +469,15 @@ MEMKIND_EXPORT void *memkind_arena_malloc(struct memkind *kind, size_t size)
     return result;
 }
 
+MEMKIND_EXPORT void memkind_arena_free(struct memkind *kind, void *ptr)
+{
+    if (ptr) {
+        unsigned int arena;
+        kind->ops->get_arena(kind, &arena, 0);
```
Comment (pbalcer) [inline src/memkind_arena.c]: assert(arena != 0);

Comment (michalbiesek) [reply src/memkind_arena.c]: Done

**Pull request merged**